miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -48,267 +47,180 @@
-      ValidationResult shortNameValidation = namingScheme.validateProjectShortName(project.getShortName());
-      if (!shortNameValidation.isValid()) {
-        throw new IOException("Cannot save project - invalid shortName: " + shortNameValidation.getMessage());
-      }
-      if (project.getId() == ProjectImpl.UNSAVED_ID) {
-        resolveMembers(project.getSecurityProfile());
-        project.setName(generateTemporaryName());
-        projectStore.save(project);
-        try {
-          project.setName(namingScheme.generateNameFor(project));
-        } catch (MisoNamingException e) {
-          throw new IOException("Cannot save Project - issue with naming scheme", e);
-        }
-        LimsUtils.validateNameOrThrow(project, namingScheme);
-      } else {
-        Project original = projectStore.get(project.getId());
-        original.setAlias(project.getAlias());
-        original.setDescription(project.getDescription());
-        original.setIssueKeys(project.getIssueKeys());
-        original.setLastUpdated(project.getLastUpdated());
-        original.setProgress(project.getProgress());
-        original.setReferenceGenome(referenceGenomeDao.getReferenceGenome(project.getReferenceGenome().getId()));
-        original.setShortName(project.getShortName());
-        for (ProjectOverview po : project.getOverviews()) {
-          if (po.getId() == ProjectOverview.UNSAVED_ID) {
-            original.getOverviews().add(po);
-          }
-        }
-        updateSecurityProfile(original.getSecurityProfile(), project.getSecurityProfile());
-        project = original;
-      }
-      project.getSecurityProfile().setProfileId(saveSecurityProfile(project.getSecurityProfile()));
-      long id = projectStore.save(project);
-      return id;
-    } else {
-      throw new IOException("No projectStore available. Check that it has been declared in the Spring config.");
-    }
+  public Collection<Project> listAllProjectsBySearch(String query) throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listBySearch(query));
   }
 
-  private void updateSecurityProfile(SecurityProfile target, SecurityProfile source) throws IOException {
-    target.setAllowAllInternal(source.isAllowAllInternal());
-    target.setOwner(source.getOwner() == null ? null : securityStore.getUserById(source.getOwner().getUserId()));
-    target.setReadGroups(loadManagedGroups(source.getReadGroups()));
-    target.setWriteGroups(loadManagedGroups(source.getWriteGroups()));
-    target.setReadUsers(loadManagedUsers(source.getReadUsers()));
-    target.setWriteUsers(loadManagedUsers(source.getWriteUsers()));
+  @Override
+  public Collection<Project> listAllProjectsWithLimit(long limit) throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listAllWithLimit(limit));
   }
 
   private Collection<Group> loadManagedGroups(Collection<Group> original) throws IOException {
     if (original == null)
       return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)